components/eventing-controller/api/v1alpha2/testing_helpers.go

```diff
@@ -0,0 +1,128 @@
+package v1alpha2
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	ProtocolSettingsContentModeBinary = "BINARY"
+	DefaultName                       = "test"
+	DefaultNamespace                  = "test-namespace"
+	DefaultSink                       = "https://svc2.test.local"
+	DefaultID                         = "id"
+	DefaultMaxInFlight                = 10
+	DefaultStatusReady                = true
+	DefaultConditions                 = []Condition{
+		{
+			Type:   ConditionSubscriptionActive,
+			Status: "true",
+		},
+		{
+			Type:   ConditionSubscribed,
+			Status: "false",
+		}}
+)
+
+// +kubebuilder:object:generate=false
```

Comment:
this is required for the `controller-gen`, because it tries make a deep copy method for all the types declared in the api/v1alpha* packages, otherwise it throws you an error

Reply:
```
$ make generate-local
controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha2:-: invalid type: func(subscription *github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha2.Subscription)
```

Reply:
For now we could also move the testing code to the unit test itself. This would fix the problem at least for now until the code needs to be reused.

Reply:
Fixed, see https://github.com/kyma-project/kyma/pull/15583#discussion_r981059655